ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +20,107 @@ export interface DropDownState {
 
 require('./dropdown.scss');
 
-const dropDownOpened = new BehaviorSubject<DropDown>(null);
-
-export class DropDown extends React.Component<DropDownProps, DropDownState> {
-    private el: HTMLDivElement;
-    private content: HTMLDivElement;
-    private subscriptions: Subscription[];
-
-    constructor(props: DropDownProps) {
-        super(props);
-        this.state = {opened: false, left: 0, top: 0};
-    }
-
-    public render() {
-        let children: React.ReactNode = null;
-        if (typeof this.props.children === 'function') {
-            if (this.state.opened) {
-                const fun = this.props.children as () => React.ReactNode;
-                children = fun();
-            }
-        } else {
-            children = this.props.children as React.ReactNode;
-        }
+const dropDownOpened = new BehaviorSubject<typeof DropDown>(null);
 
-        return (
-            <div className='argo-dropdown' ref={el => (this.el = el)}>
-                <div
-                    qe-id={this.props.qeId}
-                    className='argo-dropdown__anchor'
-                    onClick={event => {
-                        this.open();
-                        event.stopPropagation();
-                    }}>
-                    <this.props.anchor />
-                </div>
-                {ReactDOM.createPortal(
-                    <div
-                        className={classNames('argo-dropdown__content', {'opened': this.state.opened, 'is-menu': this.props.isMenu})}
-                        style={{top: this.state.top, left: this.state.left}}
-                        ref={el => (this.content = el)}>
-                        {children}
-                    </div>,
-                    document.body
-                )}
-            </div>
-        );
-    }
+export function DropDown({isMenu, anchor: Anchnor, children, qeId}: DropDownProps) {
```

Comment:
```suggestion
export function DropDown({isMenu, anchor, children, qeId}: DropDownProps) {
```

the extra type is not needed since it has an interface already. `Anchnor` also has a typo